Assets/HotmaxEditorVR/universalTransform.cs

```diff
@@ -0,0 +1,225 @@
+    }
+
+    void updateEditorMode(stateManager.editorModes value)
+    {
+        _editorMode = value;
+    }
+
+    void setOneRotationGizmoVisible(int gizmo)
+    {
+        Transform xGizmo = init.rotationGizmos.transform.Find("xRotationGizmo");
+        Transform yGizmo = init.rotationGizmos.transform.Find("yRotationGizmo");
+        Transform zGizmo = init.rotationGizmos.transform.Find("zRotationGizmo");
+
+        transforms.Add(xGizmo);
+        transforms.Add(yGizmo);
+        transforms.Add(zGizmo);
+
+        for (int i = 0; i < transforms.Count; i++)
+        {
+            transforms[i].gameObject.SetActive(false);
+        }
+        transforms[gizmo].gameObject.SetActive(true);
+    }
+
+    void setAllRotationGizmosVisible(bool setVisible)
+    {
+        Transform xGizmo = init.rotationGizmos.transform.Find("xRotationGizmo");
+        Transform yGizmo = init.rotationGizmos.transform.Find("yRotationGizmo");
+        Transform zGizmo = init.rotationGizmos.transform.Find("zRotationGizmo");
+
+        transforms.Add(xGizmo);
+        transforms.Add(yGizmo);
+        transforms.Add(zGizmo);
+
+        for (int i = 0; i < transforms.Count; i++)
+        {
+            transforms[i].gameObject.SetActive(setVisible);
+        }
+    }
+
+    void Update()
+    {
+        // -- Keep class on, but check Store to turn on functionality
+        // -- else block turns off gizmo
+        if (_editorMode == stateManager.editorModes.universalTransformMode)
+        {
+            //Telekinesis Mode 
+            if (_selectedObjectIsActive)
+            {
+                init.rotationGizmos.SetActive(false);
+
+                currentPadYPos = inputManager.device2.GetAxis().y;
+
+                Vector3 offset = inputManager.hand2.transform.forward * (distToController + scrollDistance);
+                transform.position = inputManager.hand2.transform.position + offset;
+
+                if (currentPadYPos > _initialPadYPosition)
+                {
+                    scrollDistance += scrollSpeed;
+                    _initialPadYPosition = currentPadYPos;
+                }
+                if (currentPadYPos < _initialPadYPosition)
+                {
+                    scrollDistance -= scrollSpeed;
+                    _initialPadYPosition = currentPadYPos;
+                }
+            }
+            else
+            {
+                init.rotationGizmos.SetActive(true);
+                init.rotationGizmos.transform.position = transform.position;
+                init.rotationGizmos.transform.rotation = transform.rotation;
+            }
+
+            //Rotation Mode
+            if (_rotationGizmoIsSelected)
+            {
+                currentXControllerPosition = inputManager.hand2.transform.localPosition.x;
+                currentYControllerPosition = inputManager.hand2.transform.localPosition.y;
+                float moveBuffer = 0.01f;
+
+                if (_rotationMode == stateManager.rotationModes.xRotationMode)
```

Comment:
You are planning on redoing this work with another system, right? 
There is a lot of redundancy here that could be shortened if anything were to stay.

Reply:
will refine the updated code after push